block-storage/src/main/scala/coop/rchain/blockstorage/FileLMDBIndexBlockStore.scala

```diff
@@ -0,0 +1,255 @@
+        _                            <- Sync[F].delay { blockMessageRandomAccessFile.write(blockMessageByteArray) }
+        _ <- withWriteTxn { txn =>
+              index.put(
+                txn,
+                blockHash.toDirectByteBuffer,
+                endOfFileOffset.toByteString.toDirectByteBuffer
+              )
+            }
+      } yield ()
+    )
+
+  override def checkpoint(): F[Unit] =
+    ().pure[F]
+
+  override def clear(): F[Unit] =
+    lock.withPermit(
+      for {
+        blockMessageRandomAccessFile <- blockMessageRandomAccessFileRef.get
+        _                            <- Sync[F].delay { blockMessageRandomAccessFile.setLength(0) }
+        _ <- withWriteTxn { txn =>
+              index.drop(txn)
+            }
+      } yield ()
+    )
+
+  override def close(): F[Unit] =
+    lock.withPermit(
+      for {
+        blockMessageRandomAccessFile <- blockMessageRandomAccessFileRef.get
+        _                            <- Sync[F].delay { blockMessageRandomAccessFile.close() }
+      } yield ()
+    )
+}
+
+object FileLMDBIndexBlockStore {
+  private val checkpointPattern: Regex = "([0-9]+)-([0-9]+)".r
+
+  case class Config(
+      storagePath: Path,
+      indexPath: Path,
+      checkpointsDirPath: Path,
+      mapSize: Long,
+      maxDbs: Int = 1,
+      maxReaders: Int = 126,
+      noTls: Boolean = true
+  )
+
+  private[blockstorage] case class CheckpointIndex(
+      env: Env[ByteBuffer],
+      index: Dbi[ByteBuffer]
+  )
+
+  private[blockstorage] case class Checkpoint(
+      start: Long,
+      end: Long,
+      dirPath: Path,
+      storagePath: Path,
+      indexPath: Path,
+      index: Option[WeakReference[CheckpointIndex]]
+  )
+
+  private def loadCheckpoints[F[_]: Sync: Log](checkpointsDirPath: Path): F[List[Checkpoint]] =
+    for {
+      checkpointDirectories <- Sync[F].delay {
+                                checkpointsDirPath.toFile.mkdir()
+                                Files.list(checkpointsDirPath).filter(p => Files.isDirectory(p))
+                              }
+      checkpointDirectoriesList = checkpointDirectories
+        .collect(Collectors.toList[Path])
+        .asScala
+        .toList
+      checkpoints <- checkpointDirectoriesList.flatTraverse { dirPath =>
+                      dirPath.getFileName.toString match {
+                        case checkpointPattern(start, end) =>
+                          List(
+                            Checkpoint(
+                              start.toLong,
+                              end.toLong,
+                              dirPath,
+                              dirPath.resolve("storage"),
+                              dirPath.resolve("index"),
+                              None
+                            )
+                          ).pure[F]
+                        case other =>
+                          Log[F].warn(s"Ignoring directory '$other': not a valid checkpoint name") *>
```

Comment:
shouldn't that just stop rnode working all toghether? log error and close node?

Reply:
Answered above.